test/sqllogictest/role_attributes.slt

```diff
@@ -380,3 +380,101 @@ simple conn=mz_system,user=mz_system
 ALTER SYSTEM SET enable_rbac_checks TO false;
 ----
 COMPLETE 0
+
+# Test that the attribute check apply through role membership
+
+# Parent
+
+statement ok
+ALTER ROLE joe NOCREATEROLE NOCREATEDB NOCREATECLUSTER
+
```

Comment:
Could you assert here that `joe` fails to create a cluster?

Reply:
Done, I also added asserts that `joe` fails to create/alter/drop a role and create a db.